estate/models/property_offer.py

```diff
@@ -0,0 +1,68 @@
+from odoo import models, fields, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare
+from datetime import datetime, timedelta
+
+
+class PropertyOffer(models.Model):
+    _name = "property_offer"
+    _description = "An offer for a property"
+    _order = "price DESC"
+
+    price = fields.Float()
+    status = fields.Selection([
+        ("accepted", "Accepted"),
+        ("refused", "Refused")],
+        copy=False)
+    partner_id = fields.Many2one('res.partner', required=True)
+    property_id = fields.Many2one('estate_property', required=True)
+
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(compute="_compute_deadline", inverse="_inverse_deadline")
+
+    @api.depends("validity", "create_date")
+    def _compute_deadline(self):
+        for record in self:
+            create = record.create_date or fields.Date.today()
+            # convert to date if it's a datetime
+            if isinstance(create, datetime):
+                create = create.date()
+            record.date_deadline = create + timedelta(days=record.validity)
```

Comment:
```suggestion
            record.date_deadline = (fields.Date.to_data(record.create_date) or fields.Date.today()) + timedelta(days=record.validity)
```

Python always for the handy use of `or` and `and`